### DISCUSSION TITLE
Traducción del verbo "Returns"

### DISCUSSION THREAD

**Discussion by cosmoart** [reactjs/es.reactjs.org#638] [Q&A] [answered]
Es una sección muy común, pero he visto que en algunas páginas es traducida como "Devuelve" y en otras como "Retorna", Incluso en una página está traducida como "Salidas", además, no hay una traducción recomendada en la guía de estilos.

Un ejemplo puede ser:
```md
#### Returns {/*returns*/}

The current snapshot of the store which you can use in your rendering logic.
```

**Comment by carburo** [comment]
Sí, tendremos que elegir un solo término al menos para el título de la secciones de las páginas de referencia. Yo particularmente no tengo uno preferido. De momento he postergado la decisión para luego de que terminemos de traducir las páginas. De todas formas mientras tanto sería interesante ver opiniones a favor o en contra de las distintas posibilidades.

**Reply by breadadams** [reply]
Por añadir mis 2 cents, según el contexto tener un solo término resultaría confuso o incluso incorrecto.

Lo digo pensando que dependiendo de como se esta utilizando el "_return_" por ejemplo:

- The function returns a variable
- You must return to the previous step
 
Tendría un significado (y una traducción) diferente. No obstante, tener un solo termino **para cada caso** si sería lo correcto sin duda 🙂 (que es quizás a lo que te referías).

**Reply by carburo** [reply]
Sí, la idea de escoger un solo término es específicamente para la primera acepción que indicas.

**Comment by cosmoart** [comment]
Personalmente prefiero el termino "Retorna". Aunque una votación seria de utilidad.

**Comment by NGHdeveloper** [comment]
Mi humilde opinión primaria es que "returns" debería traducirse como "devuelve" (literal), el caso particular de la palabra española "retorna" se traduce al inglés como "return to" y se interpreta en la traducción al español como "retorna a" o "retorna hacia" tomando el verbo retornar como la acepción [4 (Volver al lugar o a la situación en que se estuvo)](https://www.rae.es/drae2001/retornar) según el diccionario RAE, en cambio la traducción literal "devuelve" según el diccionario RAE en su acepción [2 (Restituir algo a quien lo tenía antes)](https://www.rae.es/drae2001/devolver#0_2) es lo que realiza una función.
Si analizamos las frases en [Google Books Ngram Viewer](https://books.google.com/ngrams/graph?content=la+funci%C3%B3n+devuelve%2Cla+funci%C3%B3n+retorna&year_start=1800&year_end=2023&corpus=es-2012&smoothing=3), observamos que en su corpus solo figura la frase "la función devuelve" en el corpus español hasta el año 2012.
Si analizamos las frases en el buscador de Google, tenemos que para ["la función retorna"](https://www.google.com/search?client=firefox-b-d&q=%22la+funci%C3%B3n+retorna%22) tiene cerca de 13.600 resultados y para ["la función devuelve"](https://www.google.com/search?q=%22la+funci%C3%B3n+devuelve%22&client=firefox-b-d&ei=2jm4ZPyWHNXY1sQP6NqL2As&ved=0ahUKEwj865aGwZuAAxVVrJUCHWjtArsQ4dUDCA4&uact=5&oq=%22la+funci%C3%B3n+devuelve%22&gs_lp=Egxnd3Mtd2l6LXNlcnAiFiJsYSBmdW5jacOzbiBkZXZ1ZWx2ZSIyBhAAGAgYHjIGEAAYCBgeMgYQABgIGB4yBhAAGAgYHjIGEAAYCBgeMgYQABgIGB4yBhAAGAgYHjIGEAAYCBgeMgYQABgIGB4yBhAAGAgYHkiWKVC7EFjeH3ACeAGQAQCYAVqgAecEqgEBOLgBA8gBAPgBAcICChAAGEcY1gQYsAPCAgYQABgHGB7CAgUQABiABMICBxAAGA0YgATCAggQABgIGAcYHuIDBBgAIEGIBgGQBgg&sclient=gws-wiz-serp) cerca de 85.200 resultados.
Esto indica, según mi razonamiento (que no sé si es el apropiado), que debería traducirse como "devuelve" y agregarla al glosario.

**Comment by carburo** [comment, marked as answer]
_Return_ ya aparece en el glosario como «devolver». Para que no haya dudas he añadido una entrada específica para el nombre de la sección (Devuelve). Si existe alguna razón para cambiar el término se puede reabrir este hilo o realizar una encuesta.
